service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/C2UploadedEventHandler.java

```diff
@@ -40,4 +44,21 @@ public void sendNotifications(final C2UploadedEvent event) {
                 eventData.getReference());
         }
     }
+
+    @EventListener
+    public void sendC2UploadedNotificationToAllocatedJudge(final C2UploadedEvent event) {
+        EventData eventData = new EventData(event);
+        CaseData caseData = mapper.convertValue(eventData.getCaseDetails().getData(), CaseData.class);
+
+        if (caseData.getAllocatedJudge() != null) {
+
+            Map<String, Object> parameters = c2UploadedEmailContentProvider.buildC2UploadNotificationForAllocatedJudge(
+                eventData.getCaseDetails());
+
+            String email = caseData.getAllocatedJudge().getJudgeEmailAddress();
```

Comment:
can `getJudgeEmailAddress()` be null at this point?

Reply:
No it will not, because before entering this event they need to enter family man + hearing details.. and before entering hearing details they need to enter allocated judge in which email is mandatory here.